benchmarks/_sources.py

```diff
@@ -51,79 +51,85 @@ class SourceFormat(Enum):
+            pyarrow.field("RELOCATION_MORTGAGE_INDICATOR", pyarrow.string()),
+            # 0-padded 2 digit ints representing categorical levels, e.g. "01" -> "Prepaid or Matured"
+            pyarrow.field("Zero_Bal_Code", pyarrow.string()),
+            pyarrow.field("ZB_DTE", pyarrow.string()),  # date
+            pyarrow.field("LAST_PAID_INSTALLMENT_DATE", pyarrow.string()),
+            pyarrow.field("FORECLOSURE_DATE", pyarrow.string()),
+            pyarrow.field("DISPOSITION_DATE", pyarrow.string()),
+            pyarrow.field("FORECLOSURE_COSTS", pyarrow.float64()),
+            pyarrow.field("PROPERTY_PRESERVATION_AND_REPAIR_COSTS", pyarrow.float64()),
+            pyarrow.field("ASSET_RECOVERY_COSTS", pyarrow.float64()),
+            pyarrow.field(
+                "MISCELLANEOUS_HOLDING_EXPENSES_AND_CREDITS", pyarrow.float64()
+            ),
+            pyarrow.field("ASSOCIATED_TAXES_FOR_HOLDING_PROPERTY", pyarrow.float64()),
+            pyarrow.field("NET_SALES_PROCEEDS", pyarrow.float64()),
+            pyarrow.field("CREDIT_ENHANCEMENT_PROCEEDS", pyarrow.float64()),
+            pyarrow.field("REPURCHASES_MAKE_WHOLE_PROCEEDS", pyarrow.float64()),
+            pyarrow.field("OTHER_FORECLOSURE_PROCEEDS", pyarrow.float64()),
+            pyarrow.field("NON_INTEREST_BEARING_UPB", pyarrow.float64()),
+            # all null
+            pyarrow.field("MI_CANCEL_FLAG", pyarrow.string()),
+            pyarrow.field("RE_PROCS_FLAG", pyarrow.string()),
+            # all null
+            pyarrow.field("LOAN_HOLDBACK_INDICATOR", pyarrow.string()),
+            pyarrow.field("SERV_IND", pyarrow.string()),
+        ]
+    )
+
+
+def nyctaxi_schema():
+    return pyarrow.schema(
+        [
+            pyarrow.field("vendor_id", pyarrow.string()),
+            pyarrow.field("pickup_datetime", pyarrow.timestamp("ns")),
+            pyarrow.field("dropoff_datetime", pyarrow.timestamp("ns")),
+            pyarrow.field("passenger_count", pyarrow.int64()),
+            pyarrow.field("trip_distance", pyarrow.float64()),
+            pyarrow.field("pickup_longitude", pyarrow.float64()),
+            pyarrow.field("pickup_latitude", pyarrow.float64()),
+            pyarrow.field("rate_code", pyarrow.int64()),
+            # int {0, 1, 2} but one row with "*"
+            pyarrow.field("store_and_fwd_flag", pyarrow.string()),
+            pyarrow.field("dropoff_longitude", pyarrow.float64()),
+            pyarrow.field("dropoff_latitude", pyarrow.float64()),
+            pyarrow.field("payment_type", pyarrow.string()),
+            pyarrow.field("fare_amount", pyarrow.float64()),
+            pyarrow.field("surcharge", pyarrow.float64()),
+            pyarrow.field("mta_tax", pyarrow.float64()),
+            pyarrow.field("tip_amount", pyarrow.float64()),
+            pyarrow.field("tolls_amount", pyarrow.float64()),
+            pyarrow.field("total_amount", pyarrow.float64()),
+        ]
+    )
```

Comment:
Aaah sneaky, but yeah this would be just like R where we do need to instantiate this not at build-time but at run-time

Reply:
It makes sense, but it never obviously failed like the R did...maybe because I used `pip install -e .` or something? It may have also been substituted with `None` instead of raising an error, which might explain some of the issues